src/miral/symbols.map

```diff
@@ -403,6 +407,7 @@ global:
     non-virtual?thunk?to?miral::WindowManagementPolicy::advise_removing_from_workspace*;
     non-virtual?thunk?to?miral::WindowManagementPolicy::advise_resize*;
     non-virtual?thunk?to?miral::WindowManagementPolicy::advise_state_change*;
+    std::swap*;
```

Comment:
We are :smile:

Reply:
@mattkae not according to @RAOF 👆

Reply:
This doesn't count then? https://github.com/canonical/mir/blob/f6296889ba81e8067d2f7b6d057f362daea36b73/include/miral/miral/window_info.h#L181

Reply:
An inline function (or, as in this case, an inline explicit function template specialization) is in a grey area. 

Because the C++ standard doesn't talk about shared libraries there's nothing to guarantee where the implementation(s) live, it would be conforming to export the implementation and use it. I have seen cases where clang does this (with default destructors IIRC)

OTOH, not publishing the symbol can only cause problems if the address of the implementation is used. (Which would be weird for `std::swap<>`)

Reply:
So would it be fair to publish the symbol to stay on the safe side in this instance?

Reply:
Fun fact: there is no way via `libclang` to know if we have an explicitly inlined function :) But we can check if a function has a definition so that _should_ do the trick

Reply:
Done :+1: